### PR TITLE
Add example filetype schema created with LinkML

### PR DESCRIPTION
**Pull request by ml-evs**
Here's an example of a filetype schema made using [LinkML](https://linkml.io/linkml/), a data modelling language that can be exported to many schema formats (JSONSchema), runtime models (pydantic) and linked data formats (RDF etc). 

This could be used to populate the registry, and also aid development of the input schema for extractor code.

Usage:

```shell
pip install -r requirements.txt # (just LinkML for now)
# validate example for MPR (currently gives an error I don't yet understand)
linkml-validate -s filetype.yml -C FileType biologic_mpr.yml
# generate a JSONSchema to be used elsewhere
gen-json-schema filetype.yml
# auto-generate pydantic code for use within Python applications
gen-pydantic filetype.yml
# generate Java class models
gen-java filetype.yml
```

### CODE REVIEW

File: filetype.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: marda_filetype_schema
+id: marda_filetype_schema
+default_range: string
+prefixes:
+  schema: http://schema.org
```
Comment (PeterKraus) [inline filetype.yml]: `schema -> schema.org`

File: filetype.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      A specific encoding of data for storage purposes, defined by a set of common characteristics and expectations, that can be assumed for all files of a given file type.
+    attributes:
+      uuid:
+        identifier: true
+        slot_uri: schema:identifier
+      name:
+        slot_uri: schema:name
+      description:
+        slot_uri: schema:description
+      vendor:
```
Comment (ml-evs) [inline filetype.yml]:
```suggestion
      primary_vendor:
```
?

```suggestion
   associated_vendors: 
      multivalued: true
````

File: filetype.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        inlined_as_list: true
+        description: >-
+          Any instrument, or class of instruments, that typically create the data encoded into such files.
+      associated_software:
+        description: >-
+          Any software (proprietary or otherwise) that produces such files.
+      subformats:
+        multivalued: true
+        range: FileType
+        inlined_as_list: true
```
Comment (ml-evs) [inline filetype.yml]:
```suggestion
      subformats:
        multivalued: true
        range: FileType
        inlined_as_list: true
        
   associated_standards:
       multivalued: true
```

File: filetype.yml (hunk 2 above)
Comment (PeterKraus) [reply filetype.yml]: I think `associated_vendors` is perhaps more descriptive.

File: filetype.yml (hunk 3 above)
Comment (PeterKraus) [reply filetype.yml]: Is the plan to keep both `subformats` and `associated_standards`? I might have misunderstood.

Comment (ml-evs) [reply filetype.yml]: No sorry, bad suggestion comment that I posted rather than losing it. Just pushing stuff now.

**Comment by ml-evs** [comment]
Still getting some LinkML errors from this (you have to do some imports within the schema to load things properly, haven't quite got the right incantation yet) but this is probably ready to look at again @PeterKraus before merging

**Comment by ml-evs** [comment]
I can perhaps add a note to the README too outlining the purpose of this demo

File: biologic_mpr.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+id: >-
+  biologic-mpr
```
Comment (PeterKraus) [inline biologic_mpr.yml]: Minor thing, but I'd prefer `biologic.mpr` as it makes the suffix a little bit more obvious.

File: biologic_mpr.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+name: >-
+  Biologic MPR file
+description: >-
+  A binary data file created by Biologic cyclers, containing time- and frequency-domain data pertaining to many electrochemical techniques, such as voltammetry, potentiometry, coulometry and impedance spectroscopy.
```
Comment (PeterKraus) [inline biologic_mpr.yml]: Question about "code" style: Should we be running those files through `black` or some other formatter on commit? Or are lines of arbitrary length, such as those descriptions, OK?

File: biologic_mpr.yml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+associated_vendors: >-
+  - Biologic
+subject: ->
```
Comment (PeterKraus) [inline biologic_mpr.yml]: I'm not clear on the syntax here, but what's the difference between `>-` and `->`?

File: biologic_mpr.yml [hunk 7]
```diff
@@ -0,0 +1,17 @@
+id: >-
+  biologic-mpr
+name: >-
+  Biologic MPR file
```
Comment (PeterKraus) [inline biologic_mpr.yml]: I think it's `BioLogic`, by the way.

**Review by PeterKraus — APPROVED** [review]
Looks good. Once you figure out how to "compile" it using linkml, I think it's good to go.

File: biologic_mpr.yml (hunk 4 above)
Comment (ml-evs) [reply biologic_mpr.yml]: My only concern here is that these IDs will probably be used for canonical URLs at some point, e.g., `https://registry.marda-alliance.org/filetypes/biologic-mpr` so I guess "." isn't URL safe

File: biologic_mpr.yml (hunk 5 above)
Comment (ml-evs) [reply biologic_mpr.yml]: I was thinking of adding `yamlfmt` via pre-commit CI at a later date, but can just do it now

File: biologic_mpr.yml (hunk 6 above)
Comment (ml-evs) [reply biologic_mpr.yml]: One is a typo, the other isn't :)

**Comment by ml-evs** [comment]
I've addressed all your issues and found the source of the bug (need to understand more what `inline_as_list` does, as it only works without it...).

I'll merge this now and we can use this is as a starting point for further discussions.

File: biologic_mpr.yml (hunk 4 above)
Comment (PeterKraus) [reply biologic_mpr.yml]: Noted.

**Pull request merged**